python/ray/autoscaler/_private/load_metrics.py

```diff
@@ -34,22 +38,51 @@ def update(self,
                resource_load,
                waiting_bundles=[],
                infeasible_bundles=[]):
-        self.resource_load_by_ip[ip] = resource_load
-        self.static_resources_by_ip[ip] = static_resources
-
-        # We are not guaranteed to have a corresponding dynamic resource for
-        # every static resource because dynamic resources are based on the
-        # available resources in the heartbeat, which does not exist if it is
-        # zero. Thus, we have to update dynamic resources here.
-        dynamic_resources_update = dynamic_resources.copy()
-        for resource_name, capacity in static_resources.items():
-            if resource_name not in dynamic_resources_update:
-                dynamic_resources_update[resource_name] = 0.0
-        self.dynamic_resources_by_ip[ip] = dynamic_resources_update
+        # If light heartbeat enabled, only resources changed will be received.
+        # We should update the changed part and compare static_resources with
+        # dynamic_resources using those updated.
+        if ray._config.light_heartbeat_enabled():
```

Comment:
Can you please add a unit test in test_autoscaler.py::TestLoadMetrics with this flag enabled? Btw, we should pass in the flag as a constructor variable instead of querying the ray config directly.

Reply:
A new test suite called `testLoadMessagesWithLightHeartbeat ` has been added in `test_autoscaler.py`. 
Also the flag(s) (actually it has two flags) has been moved out side as a parameter of `update` function in `load_metric.py`.